model/jpa/src/main/java/org/keycloak/models/jpa/RealmAdapter.java

```diff
@@ -1255,9 +1272,13 @@ public void setMasterAdminClient(ClientModel client) {
 
         for (IdentityProviderEntity entity: entities) {
             IdentityProviderModel identityProviderModel = new IdentityProviderModel();
+            IdentityProviderFactory providerFactory = KeycloakFactoryUtils.getIdentityProviderFactoryById(session, entity.getProviderId());
```

Comment:
You can do this with with the following (no need for KeycloakFactoryUtils):
`session.getFactory().getProviderFactory(IdentityProviderFactory.class, entity.getProviderId())`